pages/builds/controlling_concurrency.md.erb

```diff
@@ -2,27 +2,59 @@
+
+You can add concurrency limits to steps either through Buildkite, or your `pipeline.yml` file. Defining concurrency in a `pipeline.yml` file also requires the use of the `concurrency-group` attribute. See [Concurrency Groups](/docs/builds/controlling-concurrency#concurrency-groups) for more information.
+
+<section class="Docs__troubleshooting-note">
+    <h1>I'm seeing an error about a missing concurrency_group_id when I run my pipeline upload</h1>
+    <p>This error is caused by a missing concurrency group label. If you set a concurrency limit in a pipeline.yml file, you must also set a group label.</p>
+</section>
+
 ## Concurrency Groups
 
-Concurrency groups are labels that can be used to group together Buildkite jobs when applying concurrency limits. Any job in your organization with the same concurrency group will be subject to the concurrency limits you’ve set.
+Concurrency groups are labels that can be used to group together Buildkite jobs when applying concurrency limits. When you add a group label to a step in a `pipeline.yml` file, it will be available organsiation wide. These group labels are similar to [agent meta-data](/docs/builds/build-meta-data), and are checked at job runtime to determine which jobs are allowed to run concurrently. Although they're created on individual steps, they represent concurrent access to shared resources and can be used by other pipelines. 
 
+```yml
+steps:
+- command: "tests.sh"
+  concurrency: 1
+  concurrency_group: "tests"
+```
+
+The names of these groups should be unique, unless they're accessing a shared resource like a deployment target. You may want to ensure that order or volume sensitive operations like deploys or API access don't happen at the same time, and run in order that they were created. For example, having multiple groups called `deploy` on steps that are accessing different resources will affect which jobs are able to run at the same time.  
+
+Concurrency groups guarantee that jobs will be run in the order that they were created in. Jobs inherit the creation time of their parent. Parents of jobs can be either a build or a pipeline upload job. As pipeline uploads add more jobs to the build after it has started, the jobs that they add will inherit the creation time of the pipeline upload rather than the build.  
+
+If you have multiple pipeline uploaders within a single pipeline that each contain steps with the same concurrency group: Buildkite doesn't know about the steps inside a pipeline upload until it's running and the jobs are created. If one upload runs more quickly than another, its jobs may start running before the rest have even been created. If there are steps that rely on correct ordering, take advantage of [wait](/docs/pipelines/wait-step) or [block](/docs/pipelines/block-step) steps so that you can be sure that they will run correctly each time.
```

Comment:
I feel like we still need to mention something about this case... right? Or is it no longer a thing that can happen anymore? I remember you pushed a fix for something, but I can't remember what 😅